src/main/java/appeng/util/item/FluidList.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.item;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import com.google.common.collect.Lists;
+
+import net.minecraftforge.oredict.OreDictionary;
+
+import appeng.api.config.FuzzyMode;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.storage.data.IAEItemStack;
+import appeng.api.storage.data.IAEStack;
+import appeng.api.storage.data.IItemList;
+
+
+public final class FluidList<StackType extends IAEStack> implements IItemList<StackType>
```

Comment:
Yes, we can even go a bit further and remove the need for passing the `.class` to the constructor and let it handle by the compiler.

The whole fluidlist can now have a completely new implementation. Like do we actually need a fuzzy filtering? (are there even damagable fluids?) Can we just store every fluid into a `HashMap` or `IdentityHashMap` etc. Tons of possible optimizations here.
But for now it is just about the `ItemList`